fabric-data-attachment-api-v1/src/main/java/net/fabricmc/fabric/api/attachment/v1/AttachmentSyncPredicate.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.attachment.v1;
+
+import java.util.function.BiPredicate;
+
+import org.jetbrains.annotations.ApiStatus;
+
+import net.minecraft.server.network.ServerPlayerEntity;
+
+import net.fabricmc.fabric.impl.attachment.sync.AttachmentSyncPredicateImpl;
+
+@ApiStatus.NonExtendable
+public interface AttachmentSyncPredicate {
```

Comment:
Can this extend BiPredicate<AttachmentTarget, ServerPlayerEntity> and then there is no need for the custom function, mods would just impliment this interface?

Reply:
Coming back to this after reading some more code, I wonder if this should return a list of players that should be sent the packet, instead of being a predicate? This way all of the mostly duplicated PlayerLookup logic can possibly be moved out each target. If you dont have a clear vision of how this could work, let me know and I might have a play with it.

Reply:
Oh, thats a good point, I didnt think of that usecase.

Would it work if it was given a list or stream of possible players? In the inital login case it would only contain the one player, in the other cases it would only contain the players tracking the given target.

Reply:
I mean, it would certainly work, but that is a lot of object creation, and at that point I'm not sure there's a clear advantage over keeping the `BiPredicate` behavior and just filtering the list of all possible players with it. Which is the naive implemention, and is just a for loop.

Reply:
My main issue with the current solution is the seperation of the custom predicate, and the other standard ones. The fact the the normal ones arent implimented via the custom one likely means that the custom predicate isnt suitable.